### ISSUE TITLE
Transitioner Exception An unhandled exception of type 'System.NullReferenceException' 

### ISSUE THREAD

**Issue by Dishanx** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#777]
I am trying to create a Transitioner inside second TabItem of TabControl.
It shows me an error An unhandled exception of type 'System.NullReferenceException' occurred in MaterialDesignThemes.Wpf.dll

but it work fine with first TabItem of TabControl. Can anyone help me ?.  

```
materialDesign:Transitioner  SelectedIndex="0" AutoApplyTransitionOrigins="True">
            <local:ucHome2></local:ucHome2>
            <local:ucX></local:ucX>
            <local:ucY></local:ucY>
            <local:ucZ></local:ucZ>
        </materialDesign:Transitioner>
```

```
<TabItem >
   <TabItem.Content>
    <Grid>
    </Grid>
    </TabItem.Content>
</TabItem>
<TabItem >
   <TabItem.Content>
    <Grid>
     <local3:ucHome></local3:ucHome>
    </Grid>
    </TabItem.Content>
</TabItem>
```

**Comment by varunvasisht84** [comment]
Hello Dishanx,

I verified the behavior in Demo but putting the Tab control. Above description was not reproducible. See the screenshot below. Can you provide the sample repo?

![image](https://user-images.githubusercontent.com/17309258/29547032-6cbc97c8-8715-11e7-9c28-e68d25584adf.png)

**Comment by Dishanx** [comment]
https://github.com/Dishanx/materialDesignTransitioner1

```
 <TabItem.Content>
                <Grid>
                    <Grid.ColumnDefinitions>
                        <ColumnDefinition Width="*"/>
                    </Grid.ColumnDefinitions>
                    <Grid.RowDefinitions>
                        <RowDefinition Height="Auto"/>
                    </Grid.RowDefinitions>
                        <local3:usHome></local3:usHome> **<-- working**
                    </Grid>
            </TabItem.Content>

        </TabItem>
        <TabItem >
            <TabItem.Header>
                <StackPanel Orientation="Vertical" Width="150" Height="70" Margin="2">
                    <materialDesign:PackIcon Kind="Home" Height="24" Width="24" VerticalAlignment="Center" HorizontalAlignment="Center" />
                    <TextBlock Text="Home" Margin="0,5,0,5" VerticalAlignment="Center" HorizontalAlignment="Center" />
                </StackPanel>
            </TabItem.Header>
            <TabItem.Content>
                <Grid>
                    <Grid.ColumnDefinitions>
                        <ColumnDefinition Width="*"/>
                    </Grid.ColumnDefinitions>
                    <Grid.RowDefinitions>
                        <RowDefinition Height="Auto"/>
                    </Grid.RowDefinitions>
                       **<----- here ----- not working----**
                    </Grid>
            </TabItem.Content>
        </TabItem>
```

**Comment by Dishanx** [comment]
System.NullReferenceException: 'Object reference not set to an instance of an object.' VS 2017 and VS 2015

**Comment by Dishanx** [comment]
Hi Varun,

Do you know how to bypass the Page
```
<materialDesign:Transitioner SelectedIndex="0">
    <WizardPageOne />
    <WizardPageTwo />
    <WizardPageThree />
</materialDesign:Transitioner>

```
Based on radio button value on WizardPageOne, Want to bypass WizardPageTwo and display WizardPageThree

Radio Buttons 
1 . Show Only WizardPageTwo
2 . Show Only WizardPageThree
3. Show Both Pages

Thanks
Dishan

**Comment by varunvasisht84** [comment]
Hi Dishan,
 i am pulling in @ButchersBoy  in the discussion.
I am not sure you can achieve this the way you are doing but principally that can be done. "Transitioner" class is derived from "ItemsPanel", so you can provide an Item Source from (may be code behind Or View Model if you are using MVVM pattern). And based on some action or condition, you can clip/add/remove/insert the page and set that item as selected using SelectedItem Or SelectedIndex. I'll find some time to make a sample application for that.

**Comment by Dishanx** [comment]
Great, Thanks Varun.

**Comment by Keboo** [comment]
@Dishanx you can automatically jump to an index in the transitioner by setting the appropriate index on the Transitioner's SelectedIndex property. So what you would need to do is rather than using the Transitioner's MoveNext/MovePrevious command you would bind the relevant data to properties in your view model. In addition you would bind the SelectedIndex to a property in your view model. Then you cna implement your own commands in the view model that control the navigation.

**Comment by Dishanx** [comment]
Hi Keboo,
i tried but no luck. Can you send me a sample Code, I'm facing a problem with multiple pages (4 pages) with Next / Previous buttons and with Radio buttons to select which page combinations to show. (1st page and 4th page, 2nd page and 4th page,1st 2nd and 4th page)
Thanks

**Comment by Keboo** [comment]
@Dishanx something like this (not great code, but should explain what I was describing).
_Using MVVMLight for the ViewModelBase and the RelayCommand, but you can swap this for whatever you like._

XAML
```xaml
<Window x:Class="Trans.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:md="http://materialdesigninxaml.net/winfx/xaml/themes"
        xmlns:trans="clr-namespace:Trans"
        mc:Ignorable="d"
        Title="MainWindow" Height="350" Width="525">
    <Window.DataContext>
        <trans:MainViewModel />
    </Window.DataContext>
    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto" />
            <RowDefinition />
            <RowDefinition Height="Auto" />
        </Grid.RowDefinitions>

        <StackPanel Orientation="Horizontal">
            <RadioButton Content="All the pages" IsChecked="{Binding IsAllPages}"/>
            <RadioButton Content="Only 1st and 4th" IsChecked="{Binding IsOneAndFour}"/>
            <RadioButton Content="Only 1st, 2nd, 4th" IsChecked="{Binding IsOneTwoAndFour}"/>
        </StackPanel>

        <md:Transitioner Grid.Row="1" SelectedIndex="{Binding PageIndex}">
            <TextBlock Text="Page 1" HorizontalAlignment="Center" VerticalAlignment="Center"/>
            <TextBlock Text="Page 2" HorizontalAlignment="Center" VerticalAlignment="Center"/>
            <TextBlock Text="Page 3" HorizontalAlignment="Center" VerticalAlignment="Center"/>
            <TextBlock Text="Page 4" HorizontalAlignment="Center" VerticalAlignment="Center"/>
        </md:Transitioner>

        <Button Content="Previous" Command="{Binding PreviousPageCommand}" Grid.Row="2" HorizontalAlignment="Left" />
        <Button Content="Next" Command="{Binding NextPageCommand}" Grid.Row="2" HorizontalAlignment="Right" />
        
    </Grid>
</Window>
```

View Model
```C#
public class MainViewModel : ViewModelBase
{
    public ICommand NextPageCommand { get; }
    public ICommand PreviousPageCommand { get; }

    private int _PageIndex;
    public int PageIndex
    {
        get => _PageIndex;
        set => Set(ref _PageIndex, value);
    }

    private bool _IsAllPages = true;
    public bool IsAllPages
    {
        get => _IsAllPages;
        set => Set(ref _IsAllPages, value);
    }

    private bool _IsOneAndFour;
    public bool IsOneAndFour
    {
        get => _IsOneAndFour;
        set => Set(ref _IsOneAndFour, value);
    }

    private bool _isOneTwoAndFour;
    public bool IsOneTwoAndFour
    {
        get => _isOneTwoAndFour;
        set => Set(ref _isOneTwoAndFour, value);
    }

    public MainViewModel()
    {
        NextPageCommand = new RelayCommand(OnNext, CanNext);
        PreviousPageCommand = new RelayCommand(OnPrevious, CanPrevious);
    }

    private bool CanPrevious()
    {
        int? previousIndex = GetPreviousIndex();
        return previousIndex != null && previousIndex != PageIndex;
    }

    private void OnPrevious()
    {
        int? previousIndex = GetPreviousIndex();
        if (previousIndex != null)
        {
            PageIndex = previousIndex.Value;
        }
    }

    private bool CanNext()
    {
        int? nextIndex = GetNextIndex();
        return nextIndex != null && nextIndex != PageIndex;
    }

    private void OnNext()
    {
        int? nextIndex = GetNextIndex();
        if (nextIndex != null)
        {
            PageIndex = nextIndex.Value;
        }
    }

    private int? GetNextIndex()
    {
        if (IsAllPages)
        {
            return Math.Min(3, PageIndex + 1);
        }
        if (IsOneAndFour)
        {
            return 3;
        }
        if (IsOneTwoAndFour)
        {
            switch (PageIndex)
            {
                case 0: return 1;
                case 1:
                case 3: return 3;
            }
        }
        return null;
    }

    private int? GetPreviousIndex()
    {
        if (IsAllPages)
        {
            return Math.Max(0, PageIndex - 1);
        }
        if (IsOneAndFour)
        {
            return 0;
        }
        if (IsOneTwoAndFour)
        {
            switch (PageIndex)
            {
                case 0:
                case 1: return 0;
                case 3: return 1;
            }
        }
        return null;
    }
}
```

**Comment by Keboo** [comment]
Fixed by #1242
